b2share/modules/remotes/views.py

```diff
@@ -0,0 +1,193 @@
+                'application/json': dict_to_json_serializer,
+            },
+            default_method_media_type={
+                'GET': 'application/json',
+            },
+            default_media_type='application/json',
+            **kwargs
+        )
+
+    def get(self, path):
+        b2drop_client = session.get('b2drop_client')
+        if not b2drop_client:
+            raise UserError("B2DROP remote not initialized")
+        return b2drop_client.list(path)
+
+
+class Jobs(ContentNegotiatedMethodView):
+    view_name = 'remotes_jobs'
+
+    def __init__(self, **kwargs):
+        """Constructor."""
+        super(Jobs, self).__init__(
+            serializers={
+                'application/json': json_serializer,
+            },
+            default_method_media_type={
+                'POST': 'application/json',
+            },
+            default_media_type='application/json',
+            **kwargs
+        )
+
+    def post(self):
+        json = request.json
+        source_remote_url = json.get('source_remote_url')
+        destination_file_url = json.get('destination_file_url')
+
+        if not source_remote_url:
+            raise UserError("missing required source_remote_url parameter")
+        if not destination_file_url:
+            raise UserError("missing required destination_file_url parameter")
+
+        b2drop_urlbase = url_for('remotes.b2drop', path='', _external=True)
+        b2drop_urlbase_relative = urlparse(b2drop_urlbase).path
+        if source_remote_url.startswith(b2drop_urlbase):
+            b2drop_path = source_remote_url[len(b2drop_urlbase):]
+        elif source_remote_url.startswith(b2drop_urlbase_relative):
+            b2drop_path = source_remote_url[len(b2drop_urlbase_relative):]
+        else:
+            raise UserError("bad source_remote_url; currently only supporting b2drop remotes")
+
+        fileBucket_urlbase = url_for('invenio_files_rest.location_api', _external=True)
+        if not destination_file_url.startswith(fileBucket_urlbase):
+            raise UserError("bad destination_file_url; must point to a file bucket object")
+        destination_file_url = destination_file_url[len(fileBucket_urlbase):]
+        destination_file_url = destination_file_url.strip('/')
+        [bucket_id, key] = destination_file_url.split('/', maxsplit=1)
+        if not key or not bucket_id:
+            raise UserError("bad destination_file_url; must be a correct URL "
+                            "pointing to a file in a file bucket object")
+
+        b2drop_client = session.get('b2drop_client')
+        if not b2drop_client:
+            raise UserError("B2DROP remote not initialized")
+
+        stream = b2drop_client.make_stream_object(b2drop_path)
+        return put_file_into_bucket(bucket_id, key, stream, stream.length())
+
+
+def put_file_into_bucket(bucket_id, key, stream, content_length):
+    # TODO: refactor invenio_files_rest to have a proper API and use that one here
```

Comment:
I didn't find a direct way in invenio-files-rest a4. I did not create an issue for this.